### PR TITLE
Lite: Softmax Op Refactored

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
1:> Code refactored to become more compact and reusable
2:> Error message rectified

### CODE REVIEW

**Comment by ANSHUMAN87** [comment]
@nutsiepully : Please help conclude this PR, it is been long pending, TIA!

**Comment by ANSHUMAN87** [comment]
@nutsiepully : Please help review the PR, TIA!

**Comment by ANSHUMAN87** [comment]
@nutsiepully : please conclude this PR, TIA!

File: tensorflow/lite/kernels/activations.cc [hunk 1]
```diff
@@ -546,195 +546,78 @@ TfLiteStatus SigmoidEval(TfLiteContext* context, TfLiteNode* node) {
...
-void Softmax2DQuantizedInt8(const TfLiteTensor* input, TfLiteTensor* output,
-                            TfLiteSoftmaxParams* params, OpData* data) {
-  const int batch_size = input->dims->data[0];
-  const int input_size = input->dims->data[1];
-  SoftmaxParams op_params;
-  op_params.input_multiplier = data->input_multiplier;
-  op_params.input_left_shift = data->input_left_shift;
-  op_params.diff_min = data->diff_min;
-  optimized_integer_ops::Softmax(op_params,
-                                 GetTensorShape({batch_size, 1, 1, input_size}),
```
Comment (nutsiepully) [inline tensorflow/lite/kernels/activations.cc]: Your code essentially replaces the `GetTensorShape({batch_size, 1, 1, input_size})` call with `GetTensorShape(input)`. However they aren't equal and would result in different shapes being passed to the underlying `optimized_integer_ops::Softmax` function.

Have you verified this using the tests?

**Review by nutsiepully — COMMENTED** [review]
Thanks for the submission. Sorry for the delay on this.

I'm not sure if the refactor is going to lead to exactly the same behavior. Some of replaced calls would pass different parameters. Were you able to verify results on the tests?

File: tensorflow/lite/kernels/activations.cc (hunk 1 above)
Comment (ANSHUMAN87) [reply tensorflow/lite/kernels/activations.cc]: @nutsiepully : Yes it is verified. All the test cases pass.

Comment (nutsiepully) [reply tensorflow/lite/kernels/activations.cc]: @ANSHUMAN87 Like I mentioned the code change will lead to different parameters being passed to the underlying function. Could you please address that.

Comment (ANSHUMAN87) [reply tensorflow/lite/kernels/activations.cc]: @nutsiepully :  That's the idea for the optimization, reuse the existing parameters, and current kernel is capable of handling that. I cant see any issue in that. Would you please describe if you are foreseeing any issue in the change, Thanks!

**Comment by ANSHUMAN87** [comment]
@nutsiepully , @jianlijianli : Gentle Reminder!

File: tensorflow/lite/kernels/activations.cc [hunk 2]
```diff
@@ -748,76 +631,19 @@ TfLiteStatus SoftmaxEval(TfLiteContext* context, TfLiteNode* node) {
...
-          NumDimensions(input));
-      return kTfLiteError;
+      return SoftmaxQuantizedInt8(context, input, output, params, data);
     }
 
     default:
       context->ReportError(
-          context, "Only float32 and uint8_t are supported currently, got %s.",
+          context,
+          "Only float32, uint8_t and Int8_t are supported currently, got %s.",
```
Comment (nutsiepully) [inline tensorflow/lite/kernels/activations.cc]: nit: Please replace Int8_t with int8_t.

File: tensorflow/lite/kernels/activations.cc (hunk 1 above)
Comment (nutsiepully) [reply tensorflow/lite/kernels/activations.cc]: @ANSHUMAN87 Fair point. I didn't look into the implementation of the Softmax.

**Review by nutsiepully — APPROVED** [review]
Looks good. Thanks for the change!
